sound/soc/sof/intel/bdw.c

```diff
@@ -519,6 +519,10 @@ static int bdw_fw_ready(struct snd_sof_dev *sdev, u32 msg_id)
 		 " Firmware info: version %d:%d-%s build %d on %s:%s\n",
 		 v->major, v->minor, v->tag, v->build, v->date, v->time);
 
+	/* only copy the fw_version into debugfs at first boot */
+	if (sdev->first_boot)
+		memcpy(&sdev->fw_version, v, sizeof(*v));
+
```

Comment:
sorry, my comment above was on this line: before copying, make sure you have a valid debugfs element.

Reply:
oh, this sdev->fw_version is not a debugfs element. It is a element to sof dev to track the firmware version for onetime.
No need to check if the debugfs creation is success, the mapping is done when we want to read or open it.

Reply:
ok thanks for the clarification. let's merge when you update on top of the updated branch